packages/terra-i18n/src/I18nAggregatorPlugin.js

```diff
@@ -0,0 +1,56 @@
+import fs from 'fs';
+import path from 'path';
+import mkdirp from 'mkdirp';
+
+import i18nSupportedLanguages from './i18nSupportedLanguages';
+
+function apply(options) {
+  function getDirectories(srcPath) {
+    return fs.readdirSync(srcPath).filter(file => fs.statSync(path.join(srcPath, file)).isDirectory());
+  }
+
+  i18nSupportedLanguages.forEach((language) => {
+    const currentLanguageMessages = {};
+    getDirectories(path.resolve(options.baseDirectory, 'node_modules')).forEach((module) => {
+      const translationFile = path.resolve(options.baseDirectory, 'node_modules', module, 'translations', `${language}.json`);
+      if (fs.existsSync(translationFile)) {
+        Object.assign(currentLanguageMessages, JSON.parse(fs.readFileSync(translationFile, 'utf8')));
+      }
+    });
+
+    if (currentLanguageMessages !== {}) {
+      mkdirp(path.resolve(options.baseDirectory, 'aggregated-translations'));
+      fs.writeFileSync(path.resolve(options.baseDirectory, 'aggregated-translations', `${language}.js`),
+`import { addLocaleData } from 'react-intl';
+import localeData from 'react-intl/locale-data/${language.split('-')[0]}';
+
+addLocaleData(localeData);
+
+const messages = ${JSON.stringify(currentLanguageMessages, null, 2)};
+
+module.exports = {
+  areTranslationsLoaded: true,
+  locale: '${language}',
+  messages,
+};`,
+      );
+    }
+  });
+}
+
+module.exports = (options) => {
+  let updatedOptions = options;
+  if (updatedOptions instanceof Array) {
```

Comment:
This format was recommended on the webpack website: https://webpack.github.io/docs/plugins.html  I'm cool with simplifying it if we think that we can count on options always being passed in a certain way.